daemon/algod/api/server/v2/handlers.go

```diff
@@ -182,6 +182,64 @@ func (v2 *Handlers) GetBlock(ctx echo.Context, round uint64, params generated.Ge
 	return ctx.Blob(http.StatusOK, contentType, data)
 }
 
+// GetProof generates a Merkle proof for a transaction in a block.
+// (GET /v2/blocks/{round}/transactions/{txid}/proof)
+func (v2 *Handlers) GetProof(ctx echo.Context, round uint64, txid string, params generated.GetProofParams) error {
+	var txID transactions.Txid
+	err := txID.UnmarshalText([]byte(txid))
+	if err != nil {
+		return badRequest(ctx, err, errNoTxnSpecified, v2.Log)
+	}
+
+	ledger := v2.Node.Ledger()
+	block, _, err := ledger.BlockCert(basics.Round(round))
+	if err != nil {
+		return internalError(ctx, err, errFailedLookingUpLedger, v2.Log)
+	}
+
+	proto := config.Consensus[block.CurrentProtocol]
+	if proto.PaysetCommit != config.PaysetCommitMerkle {
+		return internalError(ctx, err, "protocol does not support Merkle proofs", v2.Log)
+	}
+
+	tree, err := block.TxnMerkleTree()
+	if err != nil {
+		return internalError(ctx, err, "building Merkle tree", v2.Log)
+	}
```

Comment:
maybe move this section into the `if txns[idx].Txn.ID() == txID {` clause since it's being used only if the transaction is found ?